tools/font-edit/twin-fedit.c

```diff
@@ -492,95 +497,102 @@ static void button(char_t *c, XButtonEvent *bev)
 
 static void play(char_t *c)
 {
-    XEvent ev;
-    char key_string[10];
-
-    XClearArea(dpy, win, 0, 0, 0, 0, True);
-    for (;;) {
-        XNextEvent(dpy, &ev);
-        switch (ev.type) {
-        case KeyPress:
-            if (XLookupString((XKeyEvent *) &ev, key_string, sizeof(key_string),
-                              0, 0) == 1) {
-                switch (key_string[0]) {
-                case 'q':
-                    return;
-                case 'c':
-                    XClearArea(dpy, ev.xkey.window, 0, 0, 0, 0, True);
+    /* Ensures that SDL only focuses on key events*/
+    SDL_StopTextInput();
+
+    SDL_Event event;
+    int quit_state = 0;
+
+    /* keep track of the selected spline */
+    cmd_t *spline = NULL;
```

Comment:
`draw_char(c)` should be called after here. It would be better if pressing the 'q' key caused the next character to be drawn immediately. First, the 'q' key is pressed and then exit the function `play()` and the spline operations are printed. Subsequently, read the next character `(c = read_char())` and then `play(c)` is executed. Therefore, in the function `play()`, there is no need to wait for `SDL_PollEvent(&event)`. This is because the 'q' key was already pressed to move to the next character in the previous turn.

Reply:
Resolve in https://github.com/sysprog21/mado/commit/7231d62d0106fb56594ddd7785efcc9ab9779b04.